src/Kestrel.Core/Internal/Http2/Http2Connection.cs

```diff
@@ -838,6 +841,11 @@ private void StartStream<TContext>(IHttpApplication<TContext> application)
                 throw new Http2StreamErrorException(_currentHeadersStream.StreamId, CoreStrings.Http2ErrorMissingMandatoryPseudoHeaderFields, Http2ErrorCode.PROTOCOL_ERROR);
             }
 
+            if (_streams.Count >= _serverSettings.MaxConcurrentStreams)
```

Comment:
Is there a potential race here? Between checking the _streams.Count and adding the _currentHeadersStream to _streams?

Reply:
We're under the state lock, so there shouldn't be.

Reply:
We also only ever process one frame at a time.

Reply:
Oh yea, I added that state lock. I just forgot to check the callsite.